### PR TITLE
Add a menu item to edit keyboard shortcuts.

### LINKED ISSUE #1611: Advertise names of existing actions for keyboard shortcuts
From the documentation page on [custom keyboard shortcuts](http://jupyter-notebook.readthedocs.io/en/latest/examples/Notebook/Custom%20Keyboard%20Shortcuts.html):

> Of course we provide name for majority of existing action so that you do not have to re-write everything, > here is for example how to bind r back to it’s initial behavior:

I could not find the list of existing actions in the documentation, and [looked up the source](https://github.com/jupyter/notebook/blob/master/notebook/static/notebook/js/keyboardmanager.js) to find the name of the action that I was looking for (`jupyter-notebook:show-command-palette`). I suggest making these names more visible, maybe right after the quoted paragraph.

### PR DESCRIPTION
**Pull request by Carreau**
Should close #1611 as advertising names will become unnecessary.

### CODE REVIEW

File: docs/source/examples/Notebook/Custom Keyboard Shortcuts.ipynb [hunk 1]
```diff
@@ -11,15 +11,33 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
+    "Starting with Jupyter Notebook 5.0, you can customize the `command` mode shortcuts from within the Notebook Application itself. \n",
+    "\n",
+    "Head to the **`Help`** menu and select the **`Edit keyboard Shortcuts`** item.\n",
+    "A dialog will guide you through the process of adding custom keyboard shortcuts.\n",
+    "\n",
+    "Keyboard shortcut set from within the Notebook Application will be persisted to your configuration file. \n",
+    "A single action may have several shortcut attached to it."
```
Comment (blink1073) [inline docs/source/examples/Notebook/Custom Keyboard Shortcuts.ipynb]: shorcuts*

**Comment by Carreau** [comment]
Done.

File: notebook/static/notebook/js/shortcuteditor.js [hunk 2]
```diff
@@ -116,7 +116,10 @@ var KeyBindingList = createClass({
             "Casing will have no effects: (e.g: `;` and `:` are the same on english keyboards)."+
             " You need to explicitelty write the `Shift` modifier.\n"+
             "Valid modifiers are `Cmd`, `Ctrl`, `Alt` ,`Meta`, `Cmdtrl`. Refer to developper docs "+
-            "for their signification depending on the platform."
+            "for their signification depending on the platform."+
```
Comment (takluyver) [inline notebook/static/notebook/js/shortcuteditor.js]: 'signification' isn't really a word; I'm not sure what you're trying to say here?

Also, on the line above, 'developer' has just one p.

File: notebook/templates/notebook.html [hunk 3]
```diff
@@ -285,7 +285,8 @@
                     <ul  id="help_menu" class="dropdown-menu">
                         {% block help %}
                         <li id="notebook_tour" title="A quick tour of the notebook user interface"><a href="#">User Interface Tour</a></li>
-                        <li id="keyboard_shortcuts" title="Opens a tooltip with all keyboard shortcuts"><a href="#">Keyboard Shortcuts</a></li>
+                        <li id="keyboard_shortcuts" title="Opens a dialog which shows all keyboard shortcuts"><a href="#">Keyboard Shortcuts</a></li>
+                        <li id="edit_keyboard_shortcuts" title="Opens a dialog allowing you to edit Keyboard shortcuts"><a href="#">Edit Keyboard Shortcuts</a></li>
```
Comment (takluyver) [inline notebook/templates/notebook.html]: Feels a bit odd for this to be in the help menu, but I guess you're right that it's where most people will see it.

Comment (minrk) [inline notebook/templates/notebook.html]: Yup. Edit or Application menu would seem to be the norm, but our Edit menu is too long, and we don't have an Application menu.

Comment (takluyver) [inline notebook/templates/notebook.html]: I've seen a bunch of different norms:

- Windows: Tools menu
- Gnome: Edit menu
- KDE: Settings menu
- OSX: Application menu

And I've seen a few applications, like Pycharm, with settings under the File menu.

File: notebook/static/notebook/js/shortcuteditor.js (hunk 2 above)
Comment (minrk) [inline notebook/static/notebook/js/shortcuteditor.js]: "For the corresponding keys"?

Comment (Carreau) [inline notebook/static/notebook/js/shortcuteditor.js]: Done, and Done.

File: notebook/templates/notebook.html (hunk 3 above)
Comment (Carreau) [inline notebook/templates/notebook.html]: I'll implement a full configuration dialog "in my free time" :-) and the we can add the classical Cog Wheel for setting somewhere.

**Pull request merged**